presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestRemoveIdentityCasts.java

```diff
@@ -0,0 +1,127 @@
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.optimizations;
+
+import com.facebook.presto.sql.planner.Plan;
+import com.facebook.presto.sql.planner.assertions.PlanAssert;
+import com.facebook.presto.sql.planner.assertions.PlanMatchPattern;
+import com.facebook.presto.testing.LocalQueryRunner;
+import com.facebook.presto.tpch.TpchConnectorFactory;
+import com.google.common.collect.ImmutableMap;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.anyTree;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.project;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+
+public class TestRemoveIdentityCasts
+{
+    private final LocalQueryRunner queryRunner;
+
+    public TestRemoveIdentityCasts()
+    {
+        this.queryRunner = new LocalQueryRunner(testSessionBuilder()
+                .setCatalog("local")
+                .setSchema("tiny")
+                .build());
+
+        queryRunner.createCatalog(queryRunner.getDefaultSession().getCatalog().get(),
+                new TpchConnectorFactory(queryRunner.getNodeManager(), 1),
+                ImmutableMap.<String, String>of());
+    }
+
+    @Test
+    public void testRemoveIdentityCastUnparameterizedColumnData()
+    {
+        @Language("SQL") String sql = "select cast(nationkey as bigint) from nation";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertNotPlan(sql, pattern);
+    }
+
+    @Test
+    public void testKeepNonIdentityCastUnparameterizedColumnData()
+    {
+        @Language("SQL") String sql = "select cast(nationkey as smallint) from nation";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertPlan(sql, pattern);
+    }
+
+    @Test
+    public void testRemoveIdentityCastParameterizedColumnData()
+    {
+        @Language("SQL") String sql = "select cast(name as varchar(25)) from nation";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertNotPlan(sql, pattern);
+    }
+
+    @Test
+    public void testKeepNonIdentityCastParameterizedColumnData()
+    {
+        @Language("SQL") String sql = "select cast(name as varchar(30)) from nation";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertPlan(sql, pattern);
+    }
+
+    @Test
+    public void testKeepNonIdentityCastDefaultParameterizedColumnData()
+    {
+        @Language("SQL") String sql = "SELECT CAST(name AS varchar) FROM nation";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertPlan(sql, pattern);
```

Comment:
I would inline `sql` and `patern` variables and merge all the test methods into a single method as they tests the same feature. Up to you.